lib/api.js

```diff
@@ -118,9 +118,9 @@ async function handleArbitraryPullRequestUpdate(context, eventData) {
       pull_number: pullRequestNumber
     });
     logger.trace("Full PR:", pullRequest);
-
+    const approvalCount = await fetchApprovalReviewCount(context, pullRequest);
     await update(context, pullRequest);
-    await merge(context, pullRequest);
+    await merge(context, pullRequest, approvalCount);
```

Comment:
Maybe instead of passing it to all the merge calls, do the fetchApprovalReviewCount call in the merge() function itself?

Reply:
To me the merge function felt like the wrong spot to fetch new data (that would also complicate mocking).
In order to not repeat the code over and over, I combined the calls to `fetchApprovalReviewCount`, `update` and `merge` into a helper function instead.